### DISCUSSION TITLE
Refresh after canvas.importSvgString

### DISCUSSION THREAD

**Discussion by hugocvx** [SVG-Edit/svgedit#529] [Q&A]
I use canvas.importSvgSgtring to add dynamically svg to existing canvas. Svg is added as can be seen into SVG code, but canvas is not refreshed. I have to open svg code editor and close to force refresh.

Using svg-editor.html
```
      var canvas = $('iframe.svgedit')[0].contentWindow.svgEditor.canvas ;
      canvas.importSvgString(
        '<svg><text x="50" y="50" fill="red">text</text></svg>'
      );
```
Canvas is not refreshed, but SVG code is ok.
`        $('iframe.svgedit')[0].contentWindow.svgEditor.canvas.getSvgString();`
Contains updated SVG with new text.

Canvas shall be refreshed with new SVG added.

**SVG-Edit environment**
- File for SVG-Edit: svg-editor.html
- Version: 4.0.1
- Protocol: http

**Desktop:**
 - OS: Windows 10
 - Browser Chrome
 - Version 70.0.3538.102

**Comment by brettz9** [comment]
SVG-Edit currently expects an SVG namespace declaration--it must be well-formed XML which is valid SVG.

Try this:

```js
      canvas.importSvgString(
        '<svg xmlns="http://www.w3.org/2000/svg"><text x="50" y="50" fill="red">text</text></svg>'
      );
```

**Comment by hugocvx** [comment]
yep, much better like this, thank you ++.
